backend/app/chat_shell/tools/builtin/knowledge_base.py

```
@@ -96,6 +96,9 @@ async def _arun(
                 f"[KnowledgeBaseTool] Searching {len(self.knowledge_base_ids)} knowledge bases with query: {query}"
             )
 
+            # Pre-fetch knowledge base names for accurate source attribution
+            kb_name_map = self._get_knowledge_base_names()
```

Ruff (0.14.10)

Local variable `kb_name_map` is assigned to but never used

Remove assignment to unused variable `kb_name_map`

(F841)